api/manifest/manifest.go

```diff
@@ -591,3 +651,112 @@ func randomString(size int) string {
 	}
 	return string(b)
 }
+
```

Comment:
this file is getting kind of large - you may want to consider putting the detection stuff in one place